tests/unit/mongo.spec.ts

```diff
@@ -2771,163 +2771,155 @@ describe.each(['36', '40', '42', '50'])(`Mongo %s translator`, version => {
+          },
+        ],
+      },
+      firstDayOfQuarter: {
+        $dateFromParts: {
+          year: { $year: '$foo' },
+          month: {
             $switch: {
               branches: [
                 { case: { $lte: [{ $divide: [{ $month: '$foo' }, 3] }, 1] }, then: 1 },
-                { case: { $lte: [{ $divide: [{ $month: '$foo' }, 3] }, 2] }, then: 2 },
-                { case: { $lte: [{ $divide: [{ $month: '$foo' }, 3] }, 3] }, then: 3 },
+                { case: { $lte: [{ $divide: [{ $month: '$foo' }, 3] }, 2] }, then: 4 },
+                { case: { $lte: [{ $divide: [{ $month: '$foo' }, 3] }, 3] }, then: 7 },
               ],
-              default: 4,
-            },
-          },
-          dayOfWeek: { $dayOfWeek: '$foo' },
-          dayOfYear: { $dayOfYear: '$foo' },
-          isoYear: { $isoWeekYear: '$foo' },
-          isoWeek: { $isoWeek: '$foo' },
-          isoDayOfWeek: { $isoDayOfWeek: '$foo' },
-          firstDayOfYear: { $dateFromParts: { year: { $year: '$foo' }, month: 1, day: 1 } },
-          firstDayOfMonth: {
-            $dateFromParts: {
-              year: { $year: '$foo' },
-              month: { $month: '$foo' },
-              day: 1,
+              default: 10,
             },
           },
-          firstDayOfWeek: {
-            // We subtract to the target date a number of days corresponding to (dayOfWeek - 1)
-            $subtract: [
-              '$foo',
-              {
-                $multiply: [{ $subtract: [{ $dayOfWeek: '$foo' }, 1] }, 24 * 60 * 60 * 1000],
-              },
-            ],
-          },
-          firstDayOfQuarter: {
-            $dateFromParts: {
-              year: { $year: '$foo' },
-              month: {
-                $switch: {
-                  branches: [
-                    { case: { $lte: [{ $divide: [{ $month: '$foo' }, 3] }, 1] }, then: 1 },
-                    { case: { $lte: [{ $divide: [{ $month: '$foo' }, 3] }, 2] }, then: 4 },
-                    { case: { $lte: [{ $divide: [{ $month: '$foo' }, 3] }, 3] }, then: 7 },
-                  ],
-                  default: 10,
-                },
-              },
-              day: 1,
-            },
+          day: 1,
+        },
+      },
+      firstDayOfIsoWeek: {
+        // We subtract to the target date a number of days corresponding to (isoDayOfWeek - 1)
+        $subtract: [
+          '$foo',
+          {
+            $multiply: [{ $subtract: [{ $isoDayOfWeek: '$foo' }, 1] }, 24 * 60 * 60 * 1000],
           },
-          firstDayOfIsoWeek: {
-            // We subtract to the target date a number of days corresponding to (isoDayOfWeek - 1)
-            $subtract: [
-              '$foo',
-              {
-                $multiply: [{ $subtract: [{ $isoDayOfWeek: '$foo' }, 1] }, 24 * 60 * 60 * 1000],
-              },
+        ],
+      },
+      previousDay: { $subtract: ['$foo', 24 * 60 * 60 * 1000] },
```

Comment:
Instead of substracting the duration (this may be false for days bwith daylight savings), prefer using the $dateSubstract operator in mongo5, which is specifically done to handle these cases
https://docs.mongodb.com/manual/reference/operator/aggregation/dateSubtract/